test/withDataProviders.test.js

```diff
@@ -146,6 +146,22 @@ test('DataProvider aborts after receiving ABORT from responseHandler', async ()
   expect(renderedMessage.textContent).toBe('')
 })
 
+test('DataProvider retries on getData timeout and accepts first returned getData result', async () => {
+  const TIMEOUT = 100
+  dataProvidersConfig({fetchTimeout: TIMEOUT})
+  // 3rd getData call returns the result first, therefore we'll expect the message to contain 'count:3'
+  const progressiveDelays = [TIMEOUT * 5, TIMEOUT * 4, TIMEOUT * 1.5, TIMEOUT * 2]
+  const {root} = renderMessageContainerApp({
+    getData: [getDataWithCount, {data: 'count:'}, progressiveDelays]
+  })
+
+  await safeDelay(TIMEOUT * 5)
+  let renderedMessage = root.querySelector('div.message p')
+  expect(renderedMessage).not.toBeNull()
+  expect(renderedMessage.textContent).toBe('count:3')
+  expect(getDataWithCount.counter).toBe(4)
```

Comment:
Hmm, why are these two numbers not equal?

Reply:
That's because the 3rd `getData` call takes `TIMEOUT * 1.5` ms to complete, so it times out and `getData` is called one more time. However the 3rd `getData` call returns the first, so that's why the actual message will be 'count:3', even though `getData` was called 4 times